src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -4404,7 +4404,10 @@ private void ImportInitBlk()
 
         private void ImportRethrow()
         {
-            EmitTrapCall();
+            // rethrows can only occur from a catch handler in which case there should be an exception slot
+            Debug.Assert(_spilledExpressions.Count > 0 && _spilledExpressions[0].Name == "ExceptionSlot");
+
+            ThrowOrRethrow(_spilledExpressions[0]);
```

Comment:
This will not do things like preserving the original stacktrace that rethrow is expected to do... . I guess good enough for now.